### ISSUE TITLE
Definitions just go until a certain point of the code

### ISSUE THREAD

**Issue by weberxw** [microsoft/pylance-release#518]
![image](https://user-images.githubusercontent.com/13738741/97095477-3a793880-1636-11eb-9436-c0026eb55652.png)
All this views are in the same file, imported through this at the top **from base_production import views**

As you can see my definitions stop in a certain point of the code, from **Deliverable FileList** down no definition is accessible.

Is there some limit based on file size or some configuration to fix this or is this a bug?

Thanks guys

**Comment by huguesv** [comment]
Hi @weberxw 

Thank you for the feedback. This does look like a bug.

The only limit that I can think of is a memory threshold during analysis where we sometimes throw away cache data to bring the memory usage back down.

Semantic colorization runs in the background with its own analysis cache, but the Go to Definition command runs in the foreground and does not share that same state.

While the screenshot clearly shows that semantic colorization stopped recognizing your view classes, **can you confirm that Go to definition command is also broken for the ones that aren't colored?**

**Could you also please try to reorder your url calls and see how that changes the result?** Like move all the Discipline url calls up above Project url calls. Mostly I'm trying to figure out if it's something in the beginning of the Discipline group that breaks things.

Another question, **does it look like this when you first open the file, before you edit anything?**

Finally, **are there any errors that appear in the Output window, under Python Language Server?** 

thanks!

**Comment by weberxw** [comment]
Hi @huguesv , thank you for the answer.

I made some tests and here are the results..

1. Its not just semantic colorization
![Captura de tela de 2020-10-27 09-51-57](https://user-images.githubusercontent.com/13738741/97304895-26525880-183b-11eb-90c5-8db7fa0dce0e.png)

2. I changed the order and the result was the same, I was able to access the ones above the limit, no matter the class
![Captura de tela de 2020-10-27 09-54-06](https://user-images.githubusercontent.com/13738741/97305171-85b06880-183b-11eb-9345-14ea62abbee7.png)

3. its not just when I open the file, I can change anything, edit, close, reopen, the definitions keep unaccessible from that point bellow

4. Thats the output log
![Captura de tela de 2020-10-27 09-58-35](https://user-images.githubusercontent.com/13738741/97305826-5ea66680-183c-11eb-95bb-8fec9eefa154.png)

Hope this can help.
TY

**Comment by huguesv** [comment]
@weberxw thank you

Can you please check the output for Python Language Server, you have to select it from the combo box on the right:
![image](https://user-images.githubusercontent.com/1696845/97338877-1e94b300-183f-11eb-8531-2610fbfda695.png)

Is this an open source project by any chance?

**Comment by weberxw** [comment]
![image](https://user-images.githubusercontent.com/13738741/97339151-003faf00-1861-11eb-9aeb-9495650f493b.png)
Ops, forgot that

No, its a private project

**Comment by huguesv** [comment]
Minimal repro:

mylib.py
```
class MyClass:
    pass
```

mytest.py
```
from . import mylib

data = [
    mylib.MyClass,
    mylib.MyClass,
    mylib.MyClass,
    mylib.MyClass,
    # copy/paste 70 or more of these...
]
```

After 64 entries, go to definition and semantic colorization on `MyClass` stops working.

It's the limit specified by `maxEntriesToUseForInference`
https://github.com/microsoft/pyright/blob/master/packages/pyright-internal/src/analyzer/typeEvaluator.ts#L607

**Comment by jakebailey** [comment]
Curious; my reading of that limit is that it should only be preventing the creation of large list/dict/set types via literals, but I it's stopping any evaluation of the node. Perhaps if the limit only stopped the addition of the item to the container but still evaluated everything, it'd work due to the evaluation happening?

I'm wondering if this means you can make a large list, then do something invalid at the end and it type check even though it's wrong.

**Comment by jakebailey** [comment]
In context, this makes sense; the screenshots in the original post hid the fact that these were appearing in a list literal.

**Comment by erictraut** [comment]
@weberxw, thanks for the bug report.
@huguesv, thanks for the investigation and repro.

I found the root cause of this bug, and I've checked in a fix.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2020.10.3, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/master/CHANGELOG.md#2020103-28-october-2020
